### PR TITLE
add rename-branch command

### PR DESCRIPTION
**Pull request by hozefaj**
solve for #535

### CODE REVIEW

File: Commands.md [hunk 1]
```diff
@@ -361,6 +362,17 @@ Does the following:
   - Executes _.git/hooks/post-release.sh_ (if present)
 
 
+## git rename-branch
+
+Rename a branch (locally and remotely).
```
Comment (spacewander) [inline Commands.md]: In my opinion, it is worth to clarify that this command doesn't really rename a remote branch. Seems it rename the upstream of local branch?

File: bin/git-rename-branch [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+test -z $1 && echo "new branch name required." 1>&2 && exit 1
+
+if [ -z $2 ]
+  then
+    $2=${git symbolic-ref --short -q HEAD}
+fi
+
+git branch -m $2 $1
+git push origin :$2
+git push --set-upstream origin $1
```
Comment (spacewander) [inline bin/git-rename-branch]: Use `branch -u/--set-upstream-to` if you just want to update the upstream.

File: Commands.md (hunk 1 above)
Comment (hemanth) [inline Commands.md]: 🌵

**Comment by spacewander** [comment]
@hozefaj 
I wonder why we need to push a branch with a comand named `rename-branch`...
Could you tell us the reason?

**Comment by hozefaj** [comment]
@spacewander Idea being to keep remote and local in sync.

File: bin/git-rename-branch (hunk 2 above)
Comment (hozefaj) [inline bin/git-rename-branch]: Not just updating, but pushing the renamed branch. `--set-upstream` allows local branch to track the remote branch.

File: Commands.md (hunk 1 above)
Comment (hozefaj) [inline Commands.md]: @spacewander Technically after renaming the local, the branch is pushed to remote. So maybe the documentation can be a bit more clear on that front. Will make the update.

- [ ] update docs clarifying `rename` functionality

**Pull request merged**